client/client-lib/src/lib.rs

```diff
@@ -81,10 +82,19 @@ pub type GatewayClient = Client<GatewayClientConfig>;
 pub type UserClient = Client<UserClientConfig>;
 
 #[derive(Debug)]
-pub struct PaymentParameters {
-    pub max_delay: u64,
-    // FIXME: change to absolute fee to avoid rounding errors
-    pub max_fee_percent: f64,
+pub enum PaymentParameters {
+    /// This invoice results in a self payment for a gateway
+    ///
+    /// Scenarios that lead to a self payment for a gateway include:
+    ///    - Payment between two user clients within the same mint
+    ///    - Payment between two user clients across mints served with the same gateway
+    Internal { hash: bitcoin_hashes::sha256::Hash },
```

Comment:
Don't we still need the amount here unless we want to parse the invoice again?

Reply:
I was thinking the amount from outgoing contract is the wrong one to use when buying preimage offer in an internal payment. Presumably, the amount from outgoing contract should be higher than the one required by the offer. When a gateway, and the difference would be the 'fee' earned by the gateway.
Would a gateway get change when they do `buy_preimage_offer(hash, amount: <AmountFromOutgoing>, rng)`?

Reply:
Turned out we needed to slightnly tweak the amount we use for coin selection within `buy_preimage_offer` : [commit](https://github.com/fedimint/fedimint/pull/495/commits/d34402b50af813fb191fcf76f455d0e6af8cd63e)